React-Projects/react-hooks/src/App.jsx

```diff
@@ -0,0 +1,84 @@
+import {
+  useState,
+  useReducer,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  createContext,
+} from "react";
+import axios from "axios";
+import "./App.css";
+import { Login } from "./Login";
+import { User } from "./User";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "INCREMENT":
+      return { count2: state.count2 + 1 };
+
+    default:
+      return state;
+  }
+};
+
+export const AppContext = createContext(null);
+function App() {
+  const [count, setCount] = useState(0);
+  const [state, dispatch] = useReducer(reducer, { count2: 0 });
+  const [data, setData] = useState("");
+  const [username, setUsername] = useState("test");
+  const inputRef = useRef(null);
+  const onClick = () => {
+    inputRef.current.focus();
+  };
+
+  useEffect(() => {
+    axios
+      .get("https://jsonplaceholder.typicode.com/comments")
+      .then((response) => {
+        console.log(response);
+        setData(response.data[0].email);
+      });
+  }, []);
+
+  useLayoutEffect(() => {
```

Comment:
@codal-kcolaco What is the use of this useLayoutEffect

Reply:
@abhi-codal useLayoutEffect runs before the browser repaints the screen. It runs even before useEffect runs. Hence, if we want any code to run before the browser repaints the screen or renders, we can use **useLayoutEffect**.